iroha/src/wsv.rs

```diff
@@ -241,7 +241,18 @@ impl WorldStateView {
                 .assets
                 .get_mut(id)
                 .ok_or_else(|| FindError::Asset(id.clone()))?;
-            f(&mut asset)
+            f(&mut asset)?;
+            // BTreeMap.retain() is not stable, therefore we use iterations
+            let mut keys_to_remove = Vec::new();
+            account.assets.iter().for_each(|(id, asset)| {
+                if asset.value.is_zero() {
+                    keys_to_remove.push(id.clone());
+                }
+            });
+            for id in keys_to_remove {
+                drop(account.assets.remove(&id));
+            }
+            Ok(())
```

Comment:
This seems like O(n) in asymptotics. I guess there should be constant ways of prunning assets (may be inside of instructions themselves?)

Reply:
This operation should be O(1). May be we can even add it in doc-comment

Reply:
Hmm, we could check only that one asset, that we (presumably) changed, you're right!

Reply:
Fixed.